tendrl/commons/objects/node_context/__init__.py

```diff
@@ -202,3 +194,95 @@ def on_change(self, attr, prev_value, current_value):
                     node_id=self.node_id,
                     integration_id=_tc.integration_id
                 )
+
+    def update_cluster_details(self, integration_id):
+        try:
+            nodes = etcd_utils.read(
+                "/clusters/%s/nodes" % integration_id
+            )
+            for node in nodes.leaves:
+                status = etcd_utils.read(
+                    "%s/NodeContext/status" % node.key
+                ).value
+                # Verify all nodes in a cluster are down
+                if status.lower() != "down":
```

Comment:
If sny of the nodes up return. right?

Reply:
yes, one node up it won't execute further